test/test_math_llama.jl

```diff
@@ -0,0 +1,118 @@
+using Llama2
+using Test
+
+@testset "math_llama.jl" begin
+    @testset "rmsnorm" begin
+        # Test 1: Typical case
+        x = Float32[1.0, 2.0, 3.0]
+        weight = Float32[1.0, 1.0, 1.0]
+        res_1 = rmsnorm(x, weight)
+        exp_1 = [0.46290955, 0.92581911, 1.38872866]
+        @test isapprox(res_1, exp_1, atol=1e-6)
+
+        # Test 2: Case with zero values
+        x = Float32[0, 0, 0, 0, 0]
+        weight = Float32[1, 1, 1, 1, 1]
+        res_2 = rmsnorm(x, weight)
+        exp_2 = [0, 0, 0, 0, 0]
+        @test isapprox(res_2, exp_2, atol=1e-6)
+
+        # Test 3: Case with negative values
+        x = Float32[-1, -2, -3, -4, -5]
+        weight = Float32[1, 1, 1, 1, 1]
+        res_3 = rmsnorm(x, weight)
+        exp_3 = [-0.30151121, -0.60302242, -0.90453362, -1.20604483, -1.50755604]
+        @test isapprox(res_3, exp_3, atol=1e-6)
+
+        # Test 4: Case with very small values
+        x = Float32[1e-6, 2e-6, 3e-6]
+        weight = Float32[1, 1, 2]
+        res_4 = rmsnorm(x, weight)
+        exp_4 = [0.00031622764, 0.00063245529, 0.00189736588]
+        @test isapprox(res_4, exp_4, atol=1e-6)
+
+        # Test 5: Edge case with single element
+        x = Float32[3.0]
+        weight = Float32[2.0]
+        res_5 = rmsnorm(x, weight)
+        exp_5 = [1.999998888889815]
+        @test res_5 ≈ exp_5 
```

Comment:
Optional test case: input empty list (should also return empty list)
(also applies to the other two functions)
In general, it is nice to have predictable behavior - either your function throws or it returns an empty list (probably the better option). Either way, you should know what will happen then.